app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,54 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = current_user.recipes
+    # @recipe = Recipe.new
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_foods = @recipe.recipe_foods
+  end
+
+  def create
+    @recipe = Recipe.new(params.require(:recipe).permit(:name, :preparation_time, :cooking_time, :description, :public, :user_id))
+    @recipe.user = current_user
+    if @recipe.save
+      redirect_to recipes_path
+    else
+      render :new
+    end
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def update
+    @recipe = Recipe.find(params[:id])
+
+    if @recipe.public.zero?
+      @recipe.update(public: 1)
+      redirect_to @recipe, notice: 'Recipe public status was successfully updated.'
+    elsif @recipe.public == 1
+      @recipe.update(public: 0)
+      redirect_to @recipe, notice: 'Recipe public status was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    authorize! :destroy, @recipe
+    if @recipe.destroy
+      redirect_to recipes_path, notice: 'Recipe deleted'
+    else
+      redirect_to recipes_path, alert: 'Recipe could not be deleted'
+    end
+  end
```

Comment:
- [ ] We should that only the user that created a food is able to delete said food for improved security of the app

Videocast
[cinnamon-2023-12-22T171604+0200.webm](https://github.com/zunairkhan811/food-recipe/assets/81932478/26905854-47c2-404d-8036-8e4925d5f4d9)

Reply:
@Munsa1 Thanks for your suggestions. I have made the changes as said by you.